### PR TITLE
ente-auth: package improvements

### PR DESCRIPTION
**Pull request by niklaskorz**
This PR addresses issues raised by @dotlambda in #339059 (`patchPhase` is currently overriden, should use `postPatch` instead) and @Aleksanaa in #352749 (don't generate multiple icon sizes, install icon to pixmaps instead). Feel free to mention anything else that might be wrong with the ente-auth package at the moment.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Review by zi3m5f — CHANGES_REQUESTED** [review]

File: pkgs/by-name/en/ente-auth/package.nix [hunk 1]
```diff
@@ -78,16 +76,7 @@ flutter324.buildFlutterApplication rec {
...
-    ICO=$out/share/icons
-
-    install -D $FAV $ICO/ente-auth.png
-    for size in 24 32 42 64 128 256 512; do
-      D=$ICO/hicolor/''${size}x''${size}/apps
-      mkdir -p $D
-      magick $FAV -resize ''${size}x''${size} $D/ente-auth.png
-    done
-
+    install -Dm644 $out/app/ente-auth/data/flutter_assets/assets/icons/auth-icon.png $out/share/pixmaps/ente-auth.png
```
[the remark this answers is not in the document]
Comment (niklaskorz) [reply pkgs/by-name/en/ente-auth/package.nix:79]: Indeed, thanks for spotting!

**Review by schnow265 — APPROVED** [review]

**Review by zi3m5f — APPROVED** [review]

**Review by schnow265 — APPROVED** [review]

**Review by gepbird — APPROVED** [review]
I haven't tested it, but the changes LGTM!

A reminder for the next update: as https://github.com/ente-io/ente/pull/3334#event-14880245912 was merged, we'll need to update the binary name and possibly the icons too.

**Comment by niklaskorz** [comment]
> I haven't tested it, but the changes LGTM!
> 
> A reminder for the next update: as [ente-io/ente#3334 (comment)](https://github.com/ente-io/ente/pull/3334#event-14880245912) was merged, we'll need to update the binary name and possibly the icons too.

Maybe we should include the rename the binary and desktop id in this PR already so back porting 4.0.3 to NixOS 24.11 will not become a breaking change?

**Comment by niklaskorz** [comment]
> Just an idea, not tested:
> 
> Another approach would be to update to [ente-io/ente#3334](https://github.com/ente-io/ente/pull/3334) and commit `e75913cb965136b4048a178c6820f4baac1e8c97`.
> 
> That PR got tested before merge I think so it wouldn't be to a completely random revision. The binary change would happen in unstable and be a bit more visible through the version `4.0.2-unstable-<date>`. It could avoid the maintenance cost of an alias script or one could keep it only in unstable till the next version.
> 
> What about the desktop file, does it also change or only `Exec`? (Didn't look as I am on mobile right now)

As I was made aware of in the Nixpkgs Matrix room, we are already in the "No breaking changes" phase of the release cycle, so either way we'll have to find a backwards-compatible solution that also allows rolling back without "losing" the data in `~/.local/share/ente_auth` (or get special permission by the release manager to still include this rename change).

**Comment by niklaskorz** [comment]
4.1.0 is out, continuing in #357503

**Pull request closed without merging**